streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -101,12 +101,21 @@
      * The type parameter controls what notion of time is used for punctuation:
      * <ul>
      *   <li>{@link PunctuationType#STREAM_TIME} - uses "stream time", which is advanced by the processing of messages
-     *   in accordance with the timestamp as extracted by the {@link TimestampExtractor} in use.
+     *   in accordance with the timestamp as extracted by the {@link TimestampExtractor} in use. The first punctuation
```

Comment:
Nit: start a new line for each sentence. (This makes diffs cleaner later on).

Maybe: `Note, that the first punctuation will be triggered by the first record that is processed.`